### PR TITLE
[NATS] initial integration

### PR DESCRIPTION
**Pull request by AdamKorcz**
This PR adds initial integration of NATS, an incubating CNCF project.

For oss-fuzz maintainers: Please await fuzzers to be merged upstream as well as NATS maintainers to add their email addresses in this PR. 

The name of the directory is nats and not nats-server, as I intend to integrate the nats-client into this directory as well.

For NATS maintainers:
@matthiashanel and @derekcollison

I have taken the liberty to add an integration application of NATS. When the fuzzers are merged in upstream, I will finish the build so it works. The fuzzers will then run continuously and all maintainers on the contact list will receive email notifications with links to bug reports, should a bug be found. If you would like to add yourself or anyone else onto the list to receive potential bug reports, please add your email address in this PR. I have added myself to be in the loop for potential breakages. This means that I will be able to see all bug reports, including security vulnerabilities.

If you have any questions or remarks about this integration, please do not hesitate to express them in this PR.

### CODE REVIEW

File: projects/nats/project.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+homepage: "https://github.com/nats-io/nats-server"
+primary_contact: ""
+auto_ccs :
+  - "adam@adalogics.com"
```
Comment (matthiashanel) [inline projects/nats/project.yaml:4]: I found the appropriate email address to use on our end, I assume it'd go here
```
- "security@nats.io"
```

**Review by matthiashanel — COMMENTED** [review]
add security@nats.io

**Review by matthiashanel — APPROVED** [review]
LGTM - thanks for adding this

**Comment by AdamKorcz** [comment]
@inferno-chromium A kind ping to let you know that this project is ready for review.

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**